### PR TITLE
Fix HIR printing of existential type #60662

### LINKED ISSUE #60662: HIR printing of `existential type` is broken
```rust
pub existential type ServeFut: Future<Output = ()>;
```

expanded with `rustc --edition 2018 -Z unpretty=hir` yields

```rust
pub existential type ServeFut : : Future<Output = ()>;
```

we should not print that extra `:`

### PR DESCRIPTION
**Pull request by Pulkit07**
This fixes https://github.com/rust-lang/rust/issues/60662

### CODE REVIEW

**Comment by hellow554** [comment]
The feature gate is missing (`#![feature(existential_type)]`)

File: src/test/ui/issues/issue-60662.rs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// compile-flags: -Z unpretty=hir
```
Comment (hellow554) [inline src/test/ui/issues/issue-60662.rs]: Feature gate is missing (`#![feature(existential_type)]`)

**Review by hellow554 — CHANGES_REQUESTED** [review]

File: src/test/ui/issues/issue-60662.rs (hunk 1 above)
Comment (Pulkit07) [reply src/test/ui/issues/issue-60662.rs]: Added that, thanks!

**Comment by alexreg** [comment]
@Pulkit07 This looks fine to me. Ping me when CI passes, and I'll approve the PR.

**Comment by alexreg** [comment]
(Maybe squash the two commits once CI passes too.)

**Review by Centril — APPROVED** [review]

**Comment by cramertj** [comment]
You'll need `// compile-pass` at the top of the test.

**Comment by alexreg** [comment]
Yep, what @cramertj said, plus squash. Then we should be good.

**Comment by alexreg** [comment]
Thanks @Pulkit07!

**Pull request merged**